### DISCUSSION TITLE
Can uno library work in winui ?

### DISCUSSION THREAD

**Discussion by DJDQfff** [unoplatform/uno#18533] [Q&A] [answered]
I have a project made with winui.
Now there is one of the pages , I want to copy same page in another unoproject.
so can i write this page into an uno library,and let winui reference the uno library?
I made a simple try.If i reference the library project visualstudio will show "project notreference";if I reference the .dll file,it will throw error that says"could not load file or assembly'uno.ui,version=255.255.255.255'""

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @DJDQfff.

Generally yes, you can use an Uno library in a WinUI project, but there are a few things to check:
- Compatibility - ensure both the Uno library and WinUI project target compatible frameworks (e.g., net6.0-windows) and use the same version of Uno.UI
- References - instead of manually referencing a .dll, add the Uno library as a project reference in your WinUI project, this avoids the errors
- NuGet Packages - make sure both projects reference the correct version of Uno.UI via NuGet, so Visual Studio manages dependencies correctly
